aws/validators.go

```diff
@@ -654,6 +655,55 @@ func validateAwsAccountId(v interface{}, k string) (ws []string, errors []error)
 	return
 }
 
+func validateKinesisAnalyticsRuntime(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	if value != kinesisanalyticsv2.RuntimeEnvironmentSql10 &&
+		value != kinesisanalyticsv2.RuntimeEnvironmentFlink16 &&
+		value != kinesisanalyticsv2.RuntimeEnvironmentFlink18 {
+
+		errors = append(errors, fmt.Errorf("'%s' is an invalid kinesisanalytics runtime. Valid runtimes are: %s", value, fmt.Sprintf("%s, %s, %s", kinesisanalyticsv2.RuntimeEnvironmentSql10, kinesisanalyticsv2.RuntimeEnvironmentFlink16,
+			kinesisanalyticsv2.RuntimeEnvironmentFlink18)))
+	}
+	return
+}
+
+func validateKineisAnalyticsMetricsLevel(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	if value != kinesisanalyticsv2.MetricsLevelApplication &&
+		value != kinesisanalyticsv2.MetricsLevelOperator &&
+		value != kinesisanalyticsv2.MetricsLevelParallelism &&
+		value != kinesisanalyticsv2.MetricsLevelTask {
+		errors = append(errors, fmt.Errorf("'%s' is an invalid kinesisanalytics metrics level. Valid values are: %s", value,
+			fmt.Sprintf("%s, %s, %s, %s", kinesisanalyticsv2.MetricsLevelApplication, kinesisanalyticsv2.MetricsLevelOperator,
+				kinesisanalyticsv2.MetricsLevelParallelism, kinesisanalyticsv2.MetricsLevelTask)))
+	}
+	return
+}
+
+func validateKinesisAnalyticsConfigurationType(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	if value != kinesisanalyticsv2.ConfigurationTypeCustom &&
+		value != kinesisanalyticsv2.ConfigurationTypeDefault {
+		errors = append(errors, fmt.Errorf("'%s' is an invalid kinesisanalytics configuration type. Valid values are: %s", value,
+			fmt.Sprintf("%s, %s", kinesisanalyticsv2.ConfigurationTypeCustom, kinesisanalyticsv2.ConfigurationTypeDefault)))
+
+	}
+	return
+}
+
+func validateLogLevel(v interface{}, k string) (ws []string, errors []error) {
```

Comment:
Rename to `validateKinesisAnalyticsV2LogLevel` to avoid potential future clashes.